src/material/legacy-select/select.ts

```diff
@@ -0,0 +1,490 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ConnectedPosition} from '@angular/cdk/overlay';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  ContentChildren,
+  Directive,
+  OnInit,
+  QueryList,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  _countGroupLabelsBeforeOption,
+  _getOptionScrollPosition,
+  MAT_OPTGROUP,
+  MAT_OPTION_PARENT_COMPONENT,
+  _MatOptionBase,
+  MatLegacyOption,
+  MatLegacyOptgroup,
+} from '@angular/material/legacy-core';
+import {MAT_SELECT_TRIGGER, _MatSelectBase} from '@angular/material/select';
+import {MatLegacyFormFieldControl} from '@angular/material/legacy-form-field';
+import {take, takeUntil} from 'rxjs/operators';
+import {matSelectAnimations} from './select-animations';
+
+/**
+ * The following style constants are necessary to save here in order
+ * to properly calculate the alignment of the selected option over
+ * the trigger element.
+ */
+
+/** The max height of the select's overlay panel. */
+export const SELECT_PANEL_MAX_HEIGHT = 256;
```

Comment:
Are we meant to prefix these exports with `LEGACY_` as well?

Reply:
My understanding was that we're only supposed to do the `Mat*` symbols, but I could be wrong.